app/app.go

```diff
@@ -277,7 +291,7 @@ func NewGaiaApp(
 		AddRoute(paramproposal.RouterKey, params.NewParamChangeProposalHandler(app.ParamsKeeper)).
 		AddRoute(distrtypes.RouterKey, distr.NewCommunityPoolSpendProposalHandler(app.DistrKeeper)).
 		AddRoute(upgradetypes.RouterKey, upgrade.NewSoftwareUpgradeProposalHandler(app.UpgradeKeeper)).
-		AddRoute(ibchost.RouterKey, ibcclient.NewClientUpdateProposalHandler(app.IBCKeeper.ClientKeeper))
+		AddRoute(ibchost.RouterKey, ibcclient.NewClientProposalHandler(app.IBCKeeper.ClientKeeper))
```

Comment:
This needs to be `ibcclienttypes.RouterKey` not `ibchost.RouterKey`. See the NOTE in the [migration docs](https://github.com/cosmos/ibc-go/blob/main/docs/migrations/ibc-migration-043.md#proposal-handler-registration)

Reply:
Thanks! updated and followed the migration guide for to integrate upgrade proposals.